src/sentry/web/api.py

```diff
@@ -332,19 +348,31 @@ def process(self, request, project, key, auth, helper, data, **kwargs):
             sender=type(self),
         )
 
-        should_filter = helper.should_filter(project, data, ip_address=remote_addr)
+        should_filter = helper.should_filter(
+            project, data, ip_address=remote_addr)
         if should_filter[0]:
+            filter_reason = should_filter[1]
```

Comment:
If you want you could change L351 to `should_filter, filter_reason = helper.should_filter(` and avoid this assignment here (and also implicitly assert the return type of `should_filter` is the 2-tuple you expect), you'd at times introduce an unnecessary local variable (`filter_reason = None`) but I think the upsides outweigh that. I'll leave that decision up to you though.